src/contract/dao/dao-app-approval.ts

```diff
@@ -1,1115 +1,1120 @@
+int 0
+byte "url"
+txna ApplicationArgs 2
+app_local_put
+int 0
+byte "url_hash"
+txna ApplicationArgs 3
+app_local_put
+txna ApplicationArgs 4
+byte ""
+==
+bnz main_l82
+int 0
+byte "hash_algo"
+txna ApplicationArgs 4
+app_local_put
+main_l74:
+txna ApplicationArgs 5
+btoi
+store 2
+txna ApplicationArgs 6
+btoi
+store 3
+txna ApplicationArgs 7
+btoi
+store 4
+txna ApplicationArgs 8
+btoi
+store 5
+load 2
+global LatestTimestamp
+>
+assert
+int 0
+byte "voting_start"
+load 2
+app_local_put
+load 3
+load 2
+>
+byte "min_duration"
+app_global_get
+load 3
+load 2
+-
+<=
+&&
+byte "max_duration"
+app_global_get
+load 3
+load 2
+-
+>=
+&&
+assert
+int 0
+byte "voting_end"
+load 3
+app_local_put
+load 4
+load 3
+>
+assert
+int 0
+byte "execute_before"
+load 4
+app_local_put
+load 5
+int 1
+==
+load 5
+int 2
+==
+||
+load 5
+int 3
+==
+||
+assert
+int 0
+byte "type"
+load 5
+app_local_put
+int 0
+byte "type"
+app_local_get
+int 1
+==
+bnz main_l81
+int 0
+byte "type"
+app_local_get
+int 2
+==
+bnz main_l80
+int 0
+byte "type"
+app_local_get
+int 3
+==
+bnz main_l78
+err
+main_l78:
+int 0
+byte "msg"
+txna ApplicationArgs 9
+app_local_put
+main_l79:
+int 0
+byte "id"
+txn TxID
+app_local_put
+int 0
+byte "executed"
+int 0
+app_local_put
+int 1
+return
+main_l80:
+int 0
+byte "from"
+txna ApplicationArgs 9
+app_local_put
+int 0
+byte "asa_id"
+txna ApplicationArgs 10
+btoi
+app_local_put
+int 0
+byte "recipient"
+txna ApplicationArgs 11
+app_local_put
+int 0
+byte "amount"
+txna ApplicationArgs 12
+btoi
+app_local_put
+b main_l79
+main_l81:
+int 0
+byte "from"
+txna ApplicationArgs 9
+app_local_put
+int 0
+byte "recipient"
+txna ApplicationArgs 10
+app_local_put
+int 0
+byte "amount"
+txna ApplicationArgs 11
+btoi
+app_local_put
+b main_l79
+main_l82:
+int 0
+byte "hash_algo"
+byte "sha256"
+app_local_put
+b main_l74
+main_l83:
+global GroupSize
+int 1
+==
+txn RekeyTo
+global ZeroAddress
+==
+&&
+txna Assets 0
+byte "gov_token_id"
+app_global_get
+==
+&&
+assert
+itxn_begin
+int axfer
+itxn_field TypeEnum
+byte "gov_token_id"
+app_global_get
+itxn_field XferAsset
+global CurrentApplicationAddress
+itxn_field AssetReceiver
+int 0
+itxn_field AssetAmount
+int 0
+itxn_field Fee
+itxn_submit
+int 1
+return
+main_l84:
+global GroupSize
+int 1
+==
+assert
+int 1
+return
+main_l85:
+int 0
+return
+main_l86:
+txna ApplicationArgs 2
+btoi
+int 0
+>
+txna ApplicationArgs 2
+btoi
+txna ApplicationArgs 3
+btoi
+<
+&&
+txna ApplicationArgs 6
+btoi
+asset_params_get AssetTotal
+store 7
+store 6
+load 7
+&&
+assert
+byte "deposit"
+txna ApplicationArgs 0
+btoi
+app_global_put
+byte "min_support"
+txna ApplicationArgs 1
+btoi
+app_global_put
+byte "min_duration"
+txna ApplicationArgs 2
+btoi
+app_global_put
+byte "max_duration"
+txna ApplicationArgs 3
+btoi
+app_global_put
+byte "url"
+txna ApplicationArgs 4
+app_global_put
+byte "dao_name"
+txna ApplicationArgs 5
+app_global_put
+byte "gov_token_id"
+txna ApplicationArgs 6
+btoi
+app_global_put
+int 1
+return
```

Comment:
I think in general we should keep the contract file (pyTEAL file) here, and use the `PyCompileOp` here to compile that contract to TEAL.

Reply:
Will create a separate task for it, as we have multiple files which should be updated